### PR TITLE
Create day73.md

### PR DESCRIPTION
**Pull request by HieuChayA4**
Create day73.md

### CODE REVIEW

File: 2022/vi/Days/day73.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+## Xây dựng Jenkins pipeline
+
+Ở bài trước, chúng ta đã triển khai Jenkins tới cụm Minikube và đã thiết lập một Jenkins Pipeline cơ bản in ra các giai đoạn của nó.
+
+Bạn cũng có thể thấy rằng có một số kịch bản mẫu cho chúng ta chạy trong quá trình tạo Jenkins Pipeline.
+
+![](../../Days/Images/Day73_CICD1.png)
+
+The first demo script is "Declarative (Kubernetes)" and you can see the stages below.
```
Comment (qmauvnt) [inline 2022/vi/Days/day73.md:19]: Remove this part

File: 2022/vi/Days/day73.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        stage('Main') {
+            steps {
+                sh 'hostname'
+            }
+        }
+    }
+}
+```
+
+You can see below the outcome of what happens when this Pipeline is run.
```
Comment (qmauvnt) [inline 2022/vi/Days/day73.md:64]: This one as well

File: 2022/vi/Days/day73.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+![](../../Days/Images/Day73_CICD6.png)
+
+Điền thông tin chi tiết tài khoản của bạn và chọn OK, nhớ rằng ID sẽ được dùng để tham chiếu khi bạn muốn gọi thông tin xác thực này. Lời khuyên của tôi ở đây là bạn nên sử dụng mã thông báo truy cập cụ thể thay vì mật khẩu.
+
+![](../../Days/Images/Day73_CICD7.png)
+
+Đối với GitHub, bạn nên sử dụng [Personal Access Token](https://vzilla.co.uk/vzilla-blog/creating-updating-your-github-personal-access-token)
+
+Tôi không thấy quy trình này khá dễ hiểu để tạo các tài khoản này, vì vậy ngay cả khi chúng ta không sử dụng, tôi muốn chia sẻ quy trình vì nó không rõ ràng từ giao diện người dùng.
```
Comment (qmauvnt) [inline 2022/vi/Days/day73.md:117]: Wording, the better expression might be

"Tôi thấy quy trình này khá khó hiểu"

File: 2022/vi/Days/day73.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+### Xây dựng Pipeline
+
+Chúng ta có thông tin xác Dockerhub được triển khai như một secret vào trong cụm Kubernetes, và chúng ta sẽ gọi nó trong giai đoạn triển khai docker đến DockerHub trong pipeline của chúng ta.
```
Comment (qmauvnt) [inline 2022/vi/Days/day73.md:121]: Wordings, is it "xác nhận" or "xác thực"?

**Review by qmauvnt — CHANGES_REQUESTED** [review]
@HieuChayA4 thanks for the contribution, well done mate.

I left some comments for the improvement, please check!

File: 2022/Days/day73.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ You might have also seen that there are some example scripts available for us to
 ![](Images/Day73_CICD1.png)
 
 The first demo script is "Declarative (Kubernetes)" and you can see the stages below.
-
+
```
Comment (qmauvnt) [inline 2022/Days/day73.md:20]: please exclude this one

File: 2022/vi/Days/day74.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+description: 90DaysOfDevOps - Hello World - Jenkinsfile App Pipeline
+tags: 'devops, 90daysofdevops, learning'
+cover_image: null
+canonical_url: null
+id: 1048744
+---
+
+## Hello World - Jenkinsfile App Pipeline
+
+Ở bài trước, chúng ta đã xây một Pipeline đơn giản trong Jenkins có thể đẩy ảnh docker từ dockerfile trong một kho lưu trữ Github công khai tới kho lưu trữ Dockerhub riêng tư.
```
Comment (qmauvnt) [inline 2022/vi/Days/day74.md:13]: docker images instead of ảnh docker

File: 2022/vi/Days/day74.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+![](../../Days/Images/Day74_CICD2.png)
+
+Hãy quay trở lại bảng điều khiển Jenkins, chúng ta sẽ tạo một pipeline mới nhưng thay vì sử kịch bản của chúng ta, chúng ta sẽ sử dụng "Pipeline script from SCM" và các tùy chọn cấu hình dưới đây.
+
+Để tham khảo, chúng ta sẽ sử dụng `https://github.com/MichaelCade/Jenkins-HelloWorld.git` làm URL của kho lưu trữ.
+
+![](../../Days/Images/Day74_CICD3.png)
+
+Ở bước này, chúng ta có thể nhấn "Lưu" và "Áp dụng", sau đó chúng ta sẽ có thể chạy thủ công Pipeline của mình để xây dựng hình ảnh Docker mới và tải lên kho DockerHub của chúng ta.
```
Comment (qmauvnt) [inline 2022/vi/Days/day74.md:41]: docker images instead of hình ảnh docker

File: 2022/vi/Days/day74.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+Tuy nhiên, tôi cũng muốn đảm bảo rằng chúng ta đã đặt lịch mỗi khi kho lưu trữ hoặc mã nguồn thay đổi sẽ kích hoạt một tiến trình xây dựng. Chúng ta có thể sử dụng webhooks hoặc đồng bộ kéo định kỳ.
+
+Điều này đáng để quan tâm vì nếu bạn đang sử dụng các tài nguyên đám mây mất phí để giữ pipeline và bạn có nhiều thay đổi tới kho lưu trữ mã nguồn bạn sẽ chịu phát sinh nhiều chi phí. Chúng ta biết rằng đây chỉ là một môi trường chạy thử đó là lí do tôi đang sử dụng lựa chọn "poll scm". (Tôi cũng tin rằng sử dụng minikube tôi sẽ thiếu đi khả năng sử dụng webhooks )
```
Comment (qmauvnt) [inline 2022/vi/Days/day74.md:44]: để giữ pipeline? not a very good expression, could you change it to something else?

Comment (qmauvnt) [reply 2022/vi/Days/day74.md:44]: tôi sẽ thiếu đi khả năng sử dụng webhooks
-> tôi sẽ không có chức năng webhooks 

is this better?

File: 2022/vi/Days/day74.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+Ở giai đoạn này, Pipeline của chúng ta chưa hề chạy và giao diện giai đoạn sẽ trông như sau.
+
+![](../../Days/Images/Day74_CICD7.png)
+
+Bây giờ hãy bấm nút "Build Now" và thẻ stage view của chúng ta sẽ hiển thị các giai đọan.
+
+![](../../Days/Images/Day74_CICD8.png)
+
+Nếu chúng ta sau đó đi đến kho DockerHub của mình, chúng ta sẽ thấy có 2 ảnh Docker mới. Chúng ta nên có một Build ID là 1 và latest vì đối với mỗi lần xây dựng mà chúng ta tạo dựa trên "Upload to DockerHub" chúng ta gửi một phiên bản bằng biến môi trường Jenkins Build_ID và chúng ta cũng phát hành một phiên bản "latest".
```
Comment (qmauvnt) [inline 2022/vi/Days/day74.md:64]: ảnh Docker -> docker image

**Review by qmauvnt — CHANGES_REQUESTED** [review]
Well done @HieuChayA4, thanks for the good contribution.

I have left some comments in the review, please check them!

**Comment by qmauvnt** [comment]
LGTM! please proceed with the new post on the blog repository @HieuChayA4

**Review by qmauvnt — CHANGES_REQUESTED** [review]
Getting rid of the change on day73 doesn't mean deleting that file 
you can check the file from the change and commit again

Well done on fixing the comments!

File: 2022/Days/day73.md [hunk 10]
```diff
@@ -223,4 +223,4 @@ The DockerHub repository I used for this demo was a private one. But in the next
 - [GitHub Actions](https://www.youtube.com/watch?v=R8_veQiYBjI)
 - [GitHub Actions CI/CD](https://www.youtube.com/watch?v=mFFXuXjVgkU)
 
-See you on [Day 74](day74.md)
+See you on [Day 74](day74.md)
```
Comment (qmauvnt) [inline 2022/Days/day73.md:226]: @HieuChayA4 hey, can you get rid of this change? This file is unrelated to this PR, so keeping it as it is is better.

**Review by qmauvnt — CHANGES_REQUESTED** [review]
thanks for the change, one last comment and we can proceed

File: 2022/Days/day73.md (hunk 10 above)
Comment (HieuChayA4) [reply 2022/Days/day73.md:226]: I'm getting trouble resolve this , i copied the the original file but i keep put this to commit :(

Comment (qmauvnt) [reply 2022/Days/day73.md:226]: you can give git rebase or git reset a try next time :D
